tests/blockchain/eth/contracts/main/miners_escrow/test_miners_escrow_additional.py

```diff
@@ -0,0 +1,372 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+import pytest
+from eth_tester.exceptions import TransactionFailed
+from web3.contract import Contract
+
+
+VALUE_FIELD = 0
+RE_STAKE_FIELD = 3
+LOCK_RE_STAKE_UNTIL_PERIOD_FIELD = 4
+
+secret = (123456).to_bytes(32, byteorder='big')
+secret2 = (654321).to_bytes(32, byteorder='big')
+
+
+@pytest.mark.slow
+def test_upgrading(testerchain, token):
+    creator = testerchain.interface.w3.eth.accounts[0]
+    miner = testerchain.interface.w3.eth.accounts[1]
+
+    secret_hash = testerchain.interface.w3.keccak(secret)
+    secret2_hash = testerchain.interface.w3.keccak(secret2)
+
+    # Deploy contract
+    contract_library_v1, _ = testerchain.interface.deploy_contract(
+        'MinersEscrow', token.address, 1, int(8e7), 4, 4, 2, 100, 1500
+    )
+    dispatcher, _ = testerchain.interface.deploy_contract('Dispatcher', contract_library_v1.address, secret_hash)
+
+    # Deploy second version of the contract
+    contract_library_v2, _ = testerchain.interface.deploy_contract(
+        'MinersEscrowV2Mock', token.address, 2, 2, 2, 2, 2, 2, 2, 2
```

Comment:
Can we have some name for these constants? Same comment for line 42.

Reply:
✔️